golang/vaultclient/createAccount_test.go

```diff
@@ -0,0 +1,141 @@
+package vaultclient
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/aws/session"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+type mockValue interface{}
+
+var (
+	mockName        = "myname"
+	mockEmail       = "email@email.com"
+	mockQuotaMax    = int64(1)
+	mockID          = "893701217479"
+	mockCanonicalID = "cdc9948f9124efae674ed122d52ce4d83d18c53ed05dcbf3765db56a051d7496"
+	mockCreateDate  = "2020-04-20T01:54:54Z"
+	mockArn         = "arn:arn:aws:iam::893701217479:/name/"
+	mockTime, _     = time.Parse(time.RFC3339, mockCreateDate)
+)
+
+func mockResponseBody(req *http.Request, t *testing.T) map[string]map[string]map[string]mockValue {
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(req.Body)
+	v, err := url.ParseQuery(string(buf.Bytes()))
+	if err != nil {
+		t.Error(err)
+	}
+	var quotaMax int64
+	if v.Get("quotaMax") != "" {
+		quotaMax, err = strconv.ParseInt(v.Get("quotaMax"), 10, 64)
+		if err != nil {
+			t.Error(err)
+		}
+	}
+	return map[string]map[string]map[string]mockValue{
+		"account": map[string]map[string]mockValue{
+			"data": map[string]mockValue{
+				"id":           mockID,
+				"emailAddress": v.Get("emailAddress"),
+				"name":         v.Get("name"),
+				"quotaMax":     quotaMax,
+				"arn":          "arn:arn:aws:iam::893701217479:/name/",
+				"canonicalId":  mockCanonicalID,
+				"createDate":   mockCreateDate,
+			},
+		},
+	}
+}
```

Comment:
`mockValue` can be of type `map[string]interface{}` (example: https://github.com/scality/zenko-operator/blob/master/pkg/controller/zenko/object_builder.go#L113) such that created the json doesn't require chaining `maps` 

so this can be written as:
```suggestion
	return mockValue{
		"account": mockValue{
			"data": mockValue{
				"id":           mockID,
				"emailAddress": v.Get("emailAddress"),
				"name":         v.Get("name"),
				"quotaMax":     quotaMax,
				"arn":          "arn:arn:aws:iam::893701217479:/name/",
				"canonicalId":  mockCanonicalID,
				"createDate":   mockCreateDate,
			},
		},
	}
}
```

Reply:
Thank you :)